File: e2e/elixir/test/sitemap_test.exs
# E2e tests for category: sitemap
defmodule E2e.SitemapTest do
  use ExUnit.Case, async: true

  describe "sitemap_basic" do
    test "Parses a standard urlset sitemap" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'urls.length' not available on result type
      # skipped: field 'has_lastmod' not available on result type
    end
  end

  describe "sitemap_compressed_gzip" do
    test "Parses a gzip-compressed sitemap file" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'urls.length' not available on result type
    end
  end

  describe "sitemap_empty" do
    test "Handles empty sitemap gracefully" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'urls.length' not available on result type
    end
  end

  describe "sitemap_from_robots_txt" do
    test "Discovers sitemap via robots.txt Sitemap directive" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'urls.length' not available on result type
    end
  end

  describe "sitemap_index" do
    test "Follows sitemap index to discover child sitemaps" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'urls.length' not available on result type
    end
  end

  describe "sitemap_lastmod_filter" do
    test "Filters sitemap URLs by lastmod date" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'urls.length' not available on result type
      # skipped: field 'has_lastmod' not available on result type
    end
  end

  describe "sitemap_only_mode" do
    test "Uses sitemap URLs exclusively without following page links" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'urls.length' not available on result type
    end
  end

  describe "sitemap_xhtml_links" do
    test "Parses sitemap with XHTML namespace alternate links" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'urls.length' not available on result type
      # skipped: field 'has_lastmod' not available on result type
    end
  end
end

File: e2e/node/tests/sitemap.test.ts
import { describe, it, expect } from 'vitest';
import { scrape } from '@kreuzberg/kreuzcrawl';

describe('sitemap', () => {
  it('sitemap_basic: Parses a standard urlset sitemap', async () => {
    const result = await scrape();
    expect(result.urls.length).toBe(4);
    expect(result.hasLastmod).toBe(true);
  });

  it('sitemap_compressed_gzip: Parses a gzip-compressed sitemap file', async () => {
    const result = await scrape();
    expect(result.urls.length).toBe(3);
  });

  it('sitemap_empty: Handles empty sitemap gracefully', async () => {
    const result = await scrape();
    expect(result.urls.length).toBe(0);
  });

  it('sitemap_from_robots_txt: Discovers sitemap via robots.txt Sitemap directive', async () => {
    const result = await scrape();
    expect(result.urls.length).toBe(4);
  });

  it('sitemap_index: Follows sitemap index to discover child sitemaps', async () => {
    const result = await scrape();
    expect(result.urls.length).toBe(3);
  });

  it('sitemap_lastmod_filter: Filters sitemap URLs by lastmod date', async () => {
    const result = await scrape();
    expect(result.urls.length).toBe(4);
    expect(result.hasLastmod).toBe(true);
  });

  it('sitemap_only_mode: Uses sitemap URLs exclusively without following page links', async () => {
    const result = await scrape();
    expect(result.urls.length).toBe(4);
  });

  it('sitemap_xhtml_links: Parses sitemap with XHTML namespace alternate links', async () => {
    const result = await scrape();
    expect(result.urls.length).toBe(2);
    expect(result.hasLastmod).toBe(false);
  });
});
